// @_davideast/stitch-mcp/dist/commands/serve/command.js
import {
  icons,
  theme
} from "../../chunk-kbtqrkwh.js";
import"../../chunk-3sfn889r.js";
import {
  exports_external
} from "../../chunk-2k7n0w2x.js";
import {
  __require
} from "../../chunk-9wyra8hs.js";

// src/commands/serve/spec.ts
var ServeOptionsSchema = exports_external.object({
  project: exports_external.string()
});

// src/commands/serve/command.ts
var command = {
  name: "serve",
  description: "Serve project HTML screens via local web server",
  requiredOptions: [
    { flags: "-p, --project <id>", description: "Project ID" }
  ],
  action: async (_args, options) => {
    try {
      const parsedOptions = ServeOptionsSchema.parse(options);
      const { ServeHandler } = await import("../../chunk-wa64nz8b.js");
      const { ServeView } = await import("../../chunk-pfyjtfex.js");
      const { StitchMCPClient } = await import("../../chunk-ezmw2j8c.js");
      const { render } = await import("../../chunk-hb3c6f6a.js");
      const React = await import("../../chunk-edp6faw2.js");
      const client = new StitchMCPClient;
      const handler = new ServeHandler(client);
      const result = await handler.execute(parsedOptions.project);
      if (!result.success) {
        console.error(theme.red(`
${icons.error} Failed: ${result.error}`));
        process.exit(1);
      }
      if (result.screens.length === 0) {
        console.log(theme.yellow(`
${icons.warning} No screens with HTML code found in this project.`));
        process.exit(0);
      }
      const createElement = React.createElement || React.default.createElement;
      const instance = render(createElement(ServeView, {
        projectId: result.projectId,
        projectTitle: result.projectTitle,
        screens: result.screens
      }));
      await instance.waitUntilExit();
      process.exit(0);
    } catch (error) {
      console.error(theme.red(`
${icons.error} Unexpected error:`), error);
      process.exit(1);
    }
  }
};
export {
  command
};

//# debugId=170DDDAD685209D664756E2164756E21
