// @_davideast/stitch-mcp/dist/commands/screens/command.js
import {
  icons,
  theme
} from "../../chunk-kbtqrkwh.js";
import"../../chunk-3sfn889r.js";
import {
  exports_external
} from "../../chunk-2k7n0w2x.js";
import {
  __require
} from "../../chunk-9wyra8hs.js";

// src/commands/screens/spec.ts
var ScreensOptionsSchema = exports_external.object({
  project: exports_external.string()
});

// src/commands/screens/command.ts
var command = {
  name: "screens",
  description: "Explore all screens in a project",
  requiredOptions: [
    { flags: "-p, --project <id>", description: "Project ID" }
  ],
  action: async (_args, options) => {
    try {
      const parsedOptions = ScreensOptionsSchema.parse(options);
      const { ScreensHandler } = await import("../../chunk-2fcdwvrm.js");
      const { ScreensView } = await import("../../chunk-nxpzt33t.js");
      const { StitchMCPClient } = await import("../../chunk-ezmw2j8c.js");
      const { render } = await import("../../chunk-hb3c6f6a.js");
      const React = await import("../../chunk-edp6faw2.js");
      const client = new StitchMCPClient;
      const handler = new ScreensHandler(client);
      const result = await handler.execute(parsedOptions.project);
      if (!result.success) {
        console.error(theme.red(`
${icons.error} Failed: ${result.error}`));
        process.exit(1);
      }
      const createElement = React.createElement || React.default.createElement;
      const instance = render(createElement(ScreensView, {
        projectId: result.projectId,
        projectTitle: result.projectTitle,
        screens: result.screens,
        client
      }));
      await instance.waitUntilExit();
      process.exit(0);
    } catch (error) {
      console.error(theme.red(`
${icons.error} Unexpected error:`), error);
      process.exit(1);
    }
  }
};
export {
  command
};

//# debugId=1A0B2AF859D0232E64756E2164756E21
