// @_davideast/stitch-mcp/dist/chunk-knbnsf6s.js
import {
  SiteBuilder,
  SiteManifest
} from "./chunk-cwkb2wbe.js";
import {
  require_jsx_dev_runtime
} from "./chunk-wz8d5vzb.js";
import {
  render_default
} from "./chunk-2cetsfw4.js";
import"./chunk-4jwmvjb4.js";
import {
  AssetGateway
} from "./chunk-7xh1y383.js";
import {
  ProjectSyncer,
  SiteService
} from "./chunk-qv44tmn6.js";
import {
  StitchMCPClient
} from "./chunk-h18jrqed.js";
import"./chunk-q4js8r0z.js";
import"./chunk-w8q7nsm7.js";
import"./chunk-nq68kghz.js";
import"./chunk-mv9ssgmx.js";
import"./chunk-nthabjd9.js";
import"./chunk-kbtqrkwh.js";
import"./chunk-3sfn889r.js";
import"./chunk-2k7n0w2x.js";
import {
  __toESM
} from "./chunk-9wyra8hs.js";

// src/commands/site/index.tsx
var jsx_dev_runtime = __toESM(require_jsx_dev_runtime(), 1);

class SiteCommandHandler {
  client;
  constructor(client) {
    this.client = client;
  }
  async execute(options) {
    const client = this.client || new StitchMCPClient;
    if (options.export) {
      const syncer = new ProjectSyncer(client);
      const remoteScreens = await syncer.fetchManifest(options.projectId);
      const uiScreens = SiteService.toUIScreens(remoteScreens);
      const siteManifest = new SiteManifest(options.projectId);
      const saved = await siteManifest.load();
      for (const screen of uiScreens) {
        const state = saved.get(screen.id);
        if (state?.status)
          screen.status = state.status;
        if (state?.route)
          screen.route = state.route;
      }
      const included = uiScreens.filter((s) => s.status === "included");
      const exportData = {
        projectId: options.projectId,
        routes: included.map((s) => ({
          screenId: s.id,
          route: s.route
        }))
      };
      console.log(JSON.stringify(exportData, null, 2));
      return;
    }
    let resultConfig = null;
    let resultHtml;
    const { waitUntilExit } = render_default(/* @__PURE__ */ jsx_dev_runtime.jsxDEV(SiteBuilder, {
      projectId: options.projectId,
      client,
      onExit: (config, html) => {
        resultConfig = config;
        resultHtml = html;
      }
    }, undefined, false, undefined, this));
    await waitUntilExit();
    if (resultConfig && resultHtml) {
      console.log("Generating site...");
      const assetGateway = new AssetGateway;
      const outputDir = options.outputDir || ".";
      await SiteService.generateSite(resultConfig, resultHtml, assetGateway, outputDir);
      console.log("Site generated successfully!");
    } else {}
  }
}
export {
  SiteCommandHandler
};

//# debugId=EFBB7A64A4433B4A64756E2164756E21
