// @_davideast/stitch-mcp/dist/chunk-2fcdwvrm.js
import"./chunk-9wyra8hs.js";

// src/commands/screens/handler.ts
class ScreensHandler {
  client;
  constructor(client) {
    this.client = client;
  }
  async execute(projectId) {
    try {
      const project = await this.client.callTool("get_project", {
        name: `projects/${projectId}`
      });
      const screensResult = await this.client.callTool("list_screens", {
        projectId
      });
      const screens = (screensResult.screens || []).map((screen) => {
        const screenId = screen.name?.split("/screens/")[1] || screen.name;
        return {
          screenId,
          title: screen.title || screenId,
          hasCode: !!screen.htmlCode?.downloadUrl,
          codeUrl: screen.htmlCode?.downloadUrl || null,
          hasImage: !!screen.screenshot?.downloadUrl
        };
      });
      screens.sort((a, b) => {
        if (a.hasCode !== b.hasCode) {
          return a.hasCode ? -1 : 1;
        }
        return a.title.localeCompare(b.title);
      });
      return {
        success: true,
        projectId,
        projectTitle: project.title || projectId,
        screens
      };
    } catch (error) {
      return {
        success: false,
        error: error.message || String(error)
      };
    }
  }
  getClient() {
    return this.client;
  }
}
export {
  ScreensHandler
};

//# debugId=826C732A3F6EF66E64756E2164756E21
