// @_davideast/stitch-mcp/dist/chunk-djhzzcgj.js
import {
  JSONRPCMessageSchema
} from "./chunk-q4js8r0z.js";
import {
  require_main
} from "./chunk-2kyqsrg2.js";
import {
  GcloudHandler,
  getStitchDir
} from "./chunk-nthabjd9.js";
import"./chunk-kbtqrkwh.js";
import"./chunk-3sfn889r.js";
import"./chunk-2k7n0w2x.js";
import {
  __toESM
} from "./chunk-9wyra8hs.js";

// node_modules/@modelcontextprotocol/sdk/dist/esm/server/stdio.js
import process2 from "node:process";

// node_modules/@modelcontextprotocol/sdk/dist/esm/shared/stdio.js
class ReadBuffer {
  append(chunk) {
    this._buffer = this._buffer ? Buffer.concat([this._buffer, chunk]) : chunk;
  }
  readMessage() {
    if (!this._buffer) {
      return null;
    }
    const index = this._buffer.indexOf(`
`);
    if (index === -1) {
      return null;
    }
    const line = this._buffer.toString("utf8", 0, index).replace(/\r$/, "");
    this._buffer = this._buffer.subarray(index + 1);
    return deserializeMessage(line);
  }
  clear() {
    this._buffer = undefined;
  }
}
function deserializeMessage(line) {
  return JSONRPCMessageSchema.parse(JSON.parse(line));
}
function serializeMessage(message) {
  return JSON.stringify(message) + `
`;
}

// node_modules/@modelcontextprotocol/sdk/dist/esm/server/stdio.js
class StdioServerTransport {
  constructor(_stdin = process2.stdin, _stdout = process2.stdout) {
    this._stdin = _stdin;
    this._stdout = _stdout;
    this._readBuffer = new ReadBuffer;
    this._started = false;
    this._ondata = (chunk) => {
      this._readBuffer.append(chunk);
      this.processReadBuffer();
    };
    this._onerror = (error) => {
      this.onerror?.(error);
    };
  }
  async start() {
    if (this._started) {
      throw new Error("StdioServerTransport already started! If using Server class, note that connect() calls start() automatically.");
    }
    this._started = true;
    this._stdin.on("data", this._ondata);
    this._stdin.on("error", this._onerror);
  }
  processReadBuffer() {
    while (true) {
      try {
        const message = this._readBuffer.readMessage();
        if (message === null) {
          break;
        }
        this.onmessage?.(message);
      } catch (error) {
        this.onerror?.(error);
      }
    }
  }
  async close() {
    this._stdin.off("data", this._ondata);
    this._stdin.off("error", this._onerror);
    const remainingDataListeners = this._stdin.listenerCount("data");
    if (remainingDataListeners === 0) {
      this._stdin.pause();
    }
    this._readBuffer.clear();
    this.onclose?.();
  }
  send(message) {
    return new Promise((resolve) => {
      const json = serializeMessage(message);
      if (this._stdout.write(json)) {
        resolve();
      } else {
        this._stdout.once("drain", resolve);
      }
    });
  }
}

// src/services/proxy/handler.ts
var import_dotenv = __toESM(require_main(), 1);
import { createWriteStream, existsSync, mkdirSync } from "node:fs";
import path from "node:path";
var deps = {
  createWriteStream,
  existsSync,
  mkdirSync,
  getStitchDir
};
var REFRESH_INTERVAL_MS = 55 * 60 * 1000;

class HttpPostTransport {
  url;
  auth;
  logger;
  onclose;
  onerror;
  onmessage;
  constructor(url, auth, logger) {
    this.url = url;
    this.auth = auth;
    this.logger = logger;
  }
  async start() {
    const project = this.auth.type === "bearer" ? this.auth.projectId : "N/A";
    this.logger(`HttpPostTransport started for ${this.url} (Auth: ${this.auth.type}, Project: ${project || "none"})`);
  }
  async send(message) {
    try {
      const msgId = message.id || "notification";
      const method = message.method || "response";
      this.logger(`Sending JSON-RPC message ID: ${msgId} Method: ${method}`);
      const headers = {
        "Content-Type": "application/json"
      };
      if (this.auth.type === "bearer") {
        const token = this.auth.tokenProvider();
        headers["Authorization"] = `Bearer ${token}`;
        if (this.auth.projectId) {
          headers["x-goog-user-project"] = this.auth.projectId;
        }
      } else {
        headers["X-Goog-Api-Key"] = this.auth.key;
      }
      const start = Date.now();
      const response = await fetch(this.url, {
        method: "POST",
        headers,
        body: JSON.stringify(message)
      });
      const duration = Date.now() - start;
      this.logger(`Response Status: ${response.status} (${duration}ms)`);
      if (!response.ok) {
        const text2 = await response.text();
        this.logger(`HTTP Error Body: ${text2}`);
        try {
          const errorJson = JSON.parse(text2);
          if (errorJson.jsonrpc === "2.0" && (errorJson.result || errorJson.error)) {
            this.logger("Forwarding error response as valid JSON-RPC");
            if (this.onmessage) {
              this.onmessage(errorJson);
            }
            return;
          }
        } catch (e) {}
        throw new Error(`HTTP ${response.status}: ${text2}`);
      }
      const text = await response.text();
      this.logger(`Response Length: ${text.length}`);
      if (!text) {
        this.logger("Empty response body received");
        return;
      }
      const data = JSON.parse(text);
      if (this.onmessage) {
        const respId = data.id || "notification";
        this.logger(`Forwarding response ID: ${respId} to local transport`);
        this.onmessage(data);
      }
    } catch (error) {
      this.logger(`Transport send error: ${error}`);
      if (this.onerror) {
        this.onerror(error instanceof Error ? error : new Error(String(error)));
      }
    }
  }
}

class ProxyHandler {
  gcloud;
  transportFactory;
  currentToken = null;
  refreshTimer = null;
  pendingToolListIds = new Set;
  constructor(gcloud = new GcloudHandler, transportFactory = () => new StdioServerTransport) {
    this.gcloud = gcloud;
    this.transportFactory = transportFactory;
  }
  async start(input) {
    let logStream = null;
    let logFile = null;
    if (input.debug) {
      try {
        const stitchDir = deps.getStitchDir();
        if (!deps.existsSync(stitchDir)) {
          deps.mkdirSync(stitchDir, { recursive: true, mode: 448 });
        }
        logFile = path.join(stitchDir, "proxy-debug.log");
        logStream = deps.createWriteStream(logFile, { flags: "a", mode: 384 });
        logStream.on("error", () => {});
      } catch (e) {}
    }
    const log = (message) => {
      if (logStream) {
        logStream.write(`[${new Date().toISOString()}] ${message}
`);
      }
    };
    if (input.debug && logFile) {
      log(`Starting ProxyHandler with debug logging enabled (File: ${logFile})`);
    }
    try {
      import_dotenv.default.config({ quiet: true });
      const apiKey = process.env.STITCH_API_KEY;
      let authConfig;
      if (apiKey) {
        log("Found STITCH_API_KEY in environment, using API Key authentication");
        authConfig = { type: "apiKey", key: apiKey };
      } else {
        await this.refreshToken();
        if (!this.currentToken) {
          log("Failed to get initial token");
          return {
            success: false,
            error: {
              code: "AUTH_REFRESH_FAILED",
              message: "Failed to retrieve initial access token",
              suggestion: 'Run "stitch-mcp init" to authenticate first',
              recoverable: false
            }
          };
        }
        const projectId = await this.gcloud.getProjectId();
        log(`Using Project ID: ${projectId}`);
        this.startRefreshTimer();
        authConfig = {
          type: "bearer",
          tokenProvider: () => this.currentToken,
          projectId: projectId ?? undefined
        };
      }
      const stitchUrl = process.env.STITCH_HOST || "https://stitch.googleapis.com/mcp";
      log(`Connecting to ${stitchUrl}`);
      const remoteTransport = new HttpPostTransport(stitchUrl, authConfig, log);
      if (input.transport !== "stdio") {
        throw new Error("Only stdio transport is supported for proxy mode currently");
      }
      const localTransport = this.transportFactory();
      localTransport.onmessage = async (message) => {
        try {
          await remoteTransport.send(message);
        } catch (e) {
          log(`Request handling error: ${e}`);
        }
      };
      remoteTransport.onmessage = async (message) => {
        try {
          await localTransport.send(message);
        } catch (e) {
          log(`Response handling error: ${e}`);
        }
      };
      remoteTransport.onerror = (error) => {
        log(`Remote transport error: ${error}`);
      };
      await remoteTransport.start();
      await localTransport.start();
      log("Access token retrieved and bridge established. Ready.");
      await new Promise((resolve) => {
        process.on("SIGINT", resolve);
        process.on("SIGTERM", resolve);
        localTransport.onclose = () => {
          log("Local connection closed");
          resolve(undefined);
        };
      });
      this.stopRefreshTimer();
      log("Proxy stopped");
      if (logStream) {
        logStream.end();
      }
      return {
        success: true,
        data: {
          status: "stopped"
        }
      };
    } catch (error) {
      this.stopRefreshTimer();
      log(`Startup failed: ${error}`);
      if (logStream) {
        logStream.end();
      }
      console.error("[Proxy] Startup failed:", error);
      return {
        success: false,
        error: {
          code: "UNKNOWN_ERROR",
          message: error instanceof Error ? error.message : String(error),
          recoverable: false
        }
      };
    }
  }
  async refreshToken() {
    const token = await this.gcloud.getAccessToken();
    if (token) {
      this.currentToken = token;
    } else {
      console.error("Failed to refresh access token");
    }
  }
  startRefreshTimer() {
    if (this.refreshTimer) {
      clearInterval(this.refreshTimer);
    }
    this.refreshTimer = setInterval(() => {
      this.refreshToken().catch((err) => console.error("Error in refresh timer:", err));
    }, REFRESH_INTERVAL_MS);
  }
  stopRefreshTimer() {
    if (this.refreshTimer) {
      clearInterval(this.refreshTimer);
      this.refreshTimer = null;
    }
  }
}

// src/commands/proxy/handler.ts
class ProxyCommandHandler {
  proxyService;
  constructor(proxyService = new ProxyHandler) {
    this.proxyService = proxyService;
  }
  async execute(input) {
    return this.proxyService.start(input);
  }
}
export {
  ProxyCommandHandler
};

//# debugId=6A0CAEEFDDA1B01664756E2164756E21
