// @_davideast/stitch-mcp/dist/chunk-nxpzt33t.js
import {
  require_jsx_dev_runtime
} from "./chunk-wz8d5vzb.js";
import {
  Box_default,
  Text,
  use_app_default,
  use_input_default
} from "./chunk-2cetsfw4.js";
import {
  require_react
} from "./chunk-4jwmvjb4.js";
import {
  StitchViteServer
} from "./chunk-7xh1y383.js";
import"./chunk-w8q7nsm7.js";
import {
  downloadText
} from "./chunk-4vxy1qce.js";
import {
  clipboardy_default
} from "./chunk-384jmtpy.js";
import"./chunk-mv9ssgmx.js";
import"./chunk-4jygt4d6.js";
import"./chunk-tz7wnw4s.js";
import"./chunk-3sfn889r.js";
import {
  __toESM
} from "./chunk-9wyra8hs.js";

// src/commands/screens/ScreensView.tsx
var import_react = __toESM(require_react(), 1);
var jsx_dev_runtime = __toESM(require_jsx_dev_runtime(), 1);
import { spawn } from "child_process";
function ScreensView({ projectId, projectTitle, screens, client }) {
  const { exit } = use_app_default();
  const [selectedIndex, setSelectedIndex] = import_react.useState(0);
  const [windowStart, setWindowStart] = import_react.useState(0);
  const [status, setStatus] = import_react.useState("");
  const [serverUrl, setServerUrl] = import_react.useState(null);
  const serverRef = import_react.useRef(null);
  const VIEW_HEIGHT = 10;
  import_react.default.useEffect(() => {
    if (selectedIndex < windowStart) {
      setWindowStart(selectedIndex);
    } else if (selectedIndex >= windowStart + VIEW_HEIGHT) {
      setWindowStart(selectedIndex - VIEW_HEIGHT + 1);
    }
  }, [selectedIndex, windowStart, VIEW_HEIGHT]);
  import_react.useEffect(() => {
    return () => {
      if (serverRef.current)
        serverRef.current.stop();
    };
  }, []);
  async function serveScreen(screen) {
    if (!screen.hasCode || !screen.codeUrl) {
      setStatus("No HTML to serve");
      return;
    }
    setStatus("Preparing server...");
    let srv = serverRef.current;
    let url = serverUrl;
    let justStarted = false;
    if (!srv) {
      srv = new StitchViteServer;
      url = await srv.start(0);
      serverRef.current = srv;
      setServerUrl(url);
      justStarted = true;
    }
    if (!url)
      return;
    try {
      const html = await downloadText(screen.codeUrl);
      const route = `/screens/${screen.screenId}`;
      srv.mount(route, html);
      const fullUrl = `${url}${route}`;
      if (justStarted) {
        const start = process.platform == "darwin" ? "open" : process.platform == "win32" ? "start" : "xdg-open";
        if (process.platform === "win32") {
          spawn("cmd", ["/c", "start", fullUrl], { detached: true, stdio: "ignore" }).unref();
        } else {
          spawn(start, [fullUrl], { detached: true, stdio: "ignore" }).unref();
        }
      } else {
        srv.navigate(fullUrl);
      }
      setStatus(`Serving at ${fullUrl}`);
    } catch (e) {
      setStatus("Error serving screen");
    }
  }
  use_input_default((input, key) => {
    if (input === "q") {
      exit();
      return;
    }
    if (key.upArrow || input === "k") {
      setSelectedIndex((prev) => Math.max(0, prev - 1));
      setStatus("");
    }
    if (key.downArrow || input === "j") {
      setSelectedIndex((prev) => Math.min(screens.length - 1, prev + 1));
      setStatus("");
    }
    if (input === "c") {
      const screen = screens[selectedIndex];
      if (screen?.hasCode && screen.codeUrl) {
        setStatus("Copying...");
        downloadText(screen.codeUrl).then((code) => {
          clipboardy_default.write(code);
          setStatus("HTML copied!");
        }).catch(() => setStatus("Failed to copy"));
      } else {
        setStatus("No HTML available");
      }
    }
    if (input === "i") {
      const screen = screens[selectedIndex];
      if (screen?.hasImage) {
        setStatus("Image copy not implemented");
      } else {
        setStatus("No image available");
      }
    }
    if (input === "s") {
      const screen = screens[selectedIndex];
      if (screen) {
        serveScreen(screen);
      }
    }
  });
  const visibleScreens = screens.slice(windowStart, windowStart + VIEW_HEIGHT);
  return /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Box_default, {
    flexDirection: "column",
    padding: 1,
    children: [
      /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
        bold: true,
        children: [
          projectTitle,
          " (",
          screens.length,
          " screens)"
        ]
      }, undefined, true, undefined, this),
      /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
        dimColor: true,
        children: [
          "projectId: ",
          projectId
        ]
      }, undefined, true, undefined, this),
      serverUrl && /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
        dimColor: true,
        children: [
          "Server: ",
          /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
            color: "green",
            children: serverUrl
          }, undefined, false, undefined, this)
        ]
      }, undefined, true, undefined, this),
      /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
        children: " "
      }, undefined, false, undefined, this),
      /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Box_default, {
        flexDirection: "column",
        borderStyle: "single",
        borderColor: "yellow",
        paddingX: 1,
        children: [
          windowStart > 0 && /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
            dimColor: true,
            children: [
              "... ",
              windowStart,
              " more above ..."
            ]
          }, undefined, true, undefined, this),
          visibleScreens.map((screen, index) => {
            const absoluteIndex = windowStart + index;
            const isSelected = absoluteIndex === selectedIndex;
            const num = String(absoluteIndex + 1).padStart(2, " ");
            const selector = isSelected ? "▸" : " ";
            return /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Box_default, {
              flexDirection: "column",
              children: [
                /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Box_default, {
                  justifyContent: "space-between",
                  children: [
                    /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Box_default, {
                      children: [
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          dimColor: true,
                          children: num
                        }, undefined, false, undefined, this),
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          color: isSelected ? "cyan" : undefined,
                          children: [
                            " ",
                            selector,
                            " "
                          ]
                        }, undefined, true, undefined, this),
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          color: isSelected ? "cyan" : undefined,
                          bold: isSelected,
                          children: screen.title.slice(0, 28)
                        }, undefined, false, undefined, this)
                      ]
                    }, undefined, true, undefined, this),
                    /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Box_default, {
                      children: [
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          dimColor: true,
                          children: "html"
                        }, undefined, false, undefined, this),
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          color: screen.hasCode ? "green" : "gray",
                          children: screen.hasCode ? "[✓]" : "[ ]"
                        }, undefined, false, undefined, this),
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          children: "  "
                        }, undefined, false, undefined, this),
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          dimColor: true,
                          children: "img"
                        }, undefined, false, undefined, this),
                        /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                          color: screen.hasImage ? "green" : "gray",
                          children: screen.hasImage ? "[✓]" : "[ ]"
                        }, undefined, false, undefined, this)
                      ]
                    }, undefined, true, undefined, this)
                  ]
                }, undefined, true, undefined, this),
                /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                  dimColor: true,
                  color: "gray",
                  children: [
                    "     screenId: ",
                    screen.screenId
                  ]
                }, undefined, true, undefined, this),
                /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
                  children: " "
                }, undefined, false, undefined, this)
              ]
            }, screen.screenId, true, undefined, this);
          }),
          windowStart + VIEW_HEIGHT < screens.length && /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
            dimColor: true,
            children: [
              "... ",
              screens.length - (windowStart + VIEW_HEIGHT),
              " more below ..."
            ]
          }, undefined, true, undefined, this)
        ]
      }, undefined, true, undefined, this),
      /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
        dimColor: true,
        children: "[c]opy html  [i]mage  [s]erve  [q]uit"
      }, undefined, false, undefined, this),
      status && /* @__PURE__ */ jsx_dev_runtime.jsxDEV(Text, {
        color: "yellow",
        children: status
      }, undefined, false, undefined, this)
    ]
  }, undefined, true, undefined, this);
}
export {
  ScreensView
};

//# debugId=D62549CBA4CF5F8B64756E2164756E21
