// @_davideast/stitch-mcp/dist/chunk-rpxnm86e.js
import {
  require_main
} from "./chunk-2kyqsrg2.js";
import {
  StitchHandler,
  createSpinner
} from "./chunk-sdp429xd.js";
import {
  ConsoleUI
} from "./chunk-z9d2xc83.js";
import {
  runSteps
} from "./chunk-f2hq6bfv.js";
import {
  GcloudHandler
} from "./chunk-nthabjd9.js";
import {
  icons,
  theme
} from "./chunk-kbtqrkwh.js";
import {
  __toESM
} from "./chunk-9wyra8hs.js";

// src/commands/doctor/handler.ts
var import_dotenv = __toESM(require_main(), 1);

// src/commands/doctor/steps/ApiKeyDetectedStep.ts
class ApiKeyDetectedStep {
  id = "api-key-detected";
  name = "Checking API Key...";
  async shouldRun(context) {
    return context.authMode === "apiKey";
  }
  async run(context) {
    const apiKey = context.apiKey;
    if (!apiKey || apiKey.trim().length === 0) {
      const message2 = "STITCH_API_KEY is set but empty";
      context.checks.push({
        name: "API Key",
        passed: false,
        message: message2,
        suggestion: "Set a valid API key in STITCH_API_KEY environment variable or in your .env file"
      });
      return { success: false, error: new Error(message2) };
    }
    const masked = apiKey.length > 7 ? `${apiKey.slice(0, 4)}...${apiKey.slice(-3)}` : "***";
    const message = `Detected (${masked})`;
    context.checks.push({
      name: "API Key",
      passed: true,
      message
    });
    return { success: true, detail: message };
  }
}

// src/commands/doctor/steps/GcloudCheckStep.ts
class GcloudCheckStep {
  id = "gcloud-check";
  name = "Checking Google Cloud CLI...";
  async shouldRun(context) {
    return context.authMode === "oauth";
  }
  async run(context) {
    const gcloudResult = await context.gcloudService.ensureInstalled({
      minVersion: "400.0.0",
      forceLocal: false
    });
    if (gcloudResult.success) {
      const message = `Installed (${gcloudResult.data.location}): v${gcloudResult.data.version}
   Path: ${gcloudResult.data.path}`;
      context.checks.push({
        name: "Google Cloud CLI",
        passed: true,
        message
      });
      return { success: true, detail: message };
    } else {
      const message = "Not found or invalid version";
      context.checks.push({
        name: "Google Cloud CLI",
        passed: false,
        message,
        suggestion: "Run: npx @_davideast/stitch-mcp init"
      });
      return { success: false, error: new Error(message) };
    }
  }
}

// src/commands/doctor/steps/AuthCheckStep.ts
class AuthCheckStep {
  id = "auth-check";
  name = "Checking user authentication...";
  async shouldRun(context) {
    return context.authMode === "oauth";
  }
  async run(context) {
    const authResult = await context.gcloudService.authenticate({ skipIfActive: true });
    if (authResult.success) {
      const message = `Authenticated: ${authResult.data.account}`;
      context.checks.push({
        name: "User Authentication",
        passed: true,
        message
      });
      return { success: true, detail: message };
    } else {
      const message = "Not authenticated";
      context.checks.push({
        name: "User Authentication",
        passed: false,
        message,
        suggestion: "Run: gcloud auth login"
      });
      return { success: false, error: new Error(message) };
    }
  }
}

// src/commands/doctor/steps/AdcCheckStep.ts
class AdcCheckStep {
  id = "adc-check";
  name = "Checking application credentials...";
  async shouldRun(context) {
    return context.authMode === "oauth";
  }
  async run(context) {
    const adcResult = await context.gcloudService.authenticateADC({ skipIfActive: true });
    if (adcResult.success) {
      const message = "Present";
      context.checks.push({
        name: "Application Credentials",
        passed: true,
        message
      });
      return { success: true, detail: message };
    } else {
      const message = "Not configured";
      context.checks.push({
        name: "Application Credentials",
        passed: false,
        message,
        suggestion: "Run: gcloud auth application-default login"
      });
      return { success: false, error: new Error(message) };
    }
  }
}

// src/commands/doctor/steps/ProjectCheckStep.ts
class ProjectCheckStep {
  id = "project-check";
  name = "Checking active project...";
  async shouldRun(context) {
    return context.authMode === "oauth";
  }
  async run(context) {
    const projectId = await context.gcloudService.getProjectId();
    if (projectId) {
      const message = `Set: ${projectId}`;
      context.checks.push({
        name: "Active Project",
        passed: true,
        message
      });
      return { success: true, detail: message };
    } else {
      const message = "No project configured";
      context.checks.push({
        name: "Active Project",
        passed: false,
        message,
        suggestion: "Run: npx @_davideast/stitch-mcp init"
      });
      return { success: false, error: new Error(message) };
    }
  }
}

// src/commands/doctor/steps/ApiKeyConnectionStep.ts
class ApiKeyConnectionStep {
  id = "api-key-connection";
  name = "Testing Stitch API...";
  async shouldRun(context) {
    if (context.authMode !== "apiKey")
      return false;
    const apiKeyCheck = context.checks.find((c) => c.name === "API Key");
    return !!apiKeyCheck && apiKeyCheck.passed;
  }
  async run(context) {
    const testResult = await context.stitchService.testConnectionWithApiKey({
      apiKey: context.apiKey
    });
    if (testResult.success) {
      const message = `Healthy (${testResult.data.statusCode})`;
      context.checks.push({
        name: "Stitch API",
        passed: true,
        message
      });
      return { success: true, detail: message };
    } else {
      const message = testResult.error.message;
      context.checks.push({
        name: "Stitch API",
        passed: false,
        message,
        suggestion: testResult.error.suggestion,
        details: testResult.error.details
      });
      return { success: false, error: new Error(message) };
    }
  }
}

// src/commands/doctor/steps/ApiCheckStep.ts
class ApiCheckStep {
  id = "api-check";
  name = "Testing Stitch API...";
  async shouldRun(context) {
    if (context.authMode !== "oauth")
      return false;
    const projectCheck = context.checks.find((c) => c.name === "Active Project");
    return !!projectCheck && projectCheck.passed;
  }
  async run(context) {
    const projectId = await context.gcloudService.getProjectId();
    const accessToken = await context.gcloudService.getAccessToken();
    if (projectId && accessToken) {
      const testResult = await context.stitchService.testConnection({
        projectId,
        accessToken
      });
      if (testResult.success) {
        const message = `Healthy (${testResult.data.statusCode})`;
        context.checks.push({
          name: "Stitch API",
          passed: true,
          message
        });
        return { success: true, detail: message };
      } else {
        const message = testResult.error.message;
        context.checks.push({
          name: "Stitch API",
          passed: false,
          message,
          suggestion: testResult.error.suggestion,
          details: testResult.error.details
        });
        return { success: false, error: new Error(message) };
      }
    } else {
      const message = "Could not obtain access token";
      context.checks.push({
        name: "Stitch API",
        passed: false,
        message,
        suggestion: "Re-run authentication"
      });
      return { success: false, error: new Error(message) };
    }
  }
}

// src/commands/doctor/handler.ts
class DoctorHandler {
  gcloudService;
  stitchService;
  steps;
  ui;
  constructor(gcloudService = new GcloudHandler, stitchService = new StitchHandler, ui) {
    this.gcloudService = gcloudService;
    this.stitchService = stitchService;
    this.ui = ui || new ConsoleUI;
    this.steps = [
      new ApiKeyDetectedStep,
      new GcloudCheckStep,
      new AuthCheckStep,
      new AdcCheckStep,
      new ProjectCheckStep,
      new ApiKeyConnectionStep,
      new ApiCheckStep
    ];
  }
  async execute(input) {
    import_dotenv.default.config({ quiet: true });
    const apiKey = process.env.STITCH_API_KEY;
    const context = {
      input,
      ui: this.ui,
      gcloudService: this.gcloudService,
      stitchService: this.stitchService,
      authMode: apiKey ? "apiKey" : "oauth",
      apiKey: apiKey || undefined,
      checks: []
    };
    console.log(`
${theme.blue("Stitch Doctor")}
`);
    try {
      let spinner;
      await runSteps(this.steps, context, {
        onBeforeStep: (step) => {
          spinner = createSpinner();
          spinner.start(step.name);
        },
        onAfterStep: (_step, result) => {
          if (result.success) {
            spinner.succeed(result.detail || "Passed");
          } else {
            spinner.fail(result.error?.message || "Failed");
          }
          return false;
        }
      });
      const allPassed = context.checks.every((c) => c.passed);
      console.log(`
${theme.blue("─".repeat(60))}
`);
      console.log(theme.blue(`Health Check Summary
`));
      for (const check of context.checks) {
        const icon = check.passed ? theme.green(icons.success) : theme.red(icons.error);
        console.log(`${icon} ${check.name}: ${check.message}`);
        if (check.suggestion && !check.passed) {
          console.log(theme.gray(`  → ${check.suggestion}`));
        }
      }
      if (input.verbose) {
        const failedChecksWithDetails = context.checks.filter((c) => !c.passed && c.details);
        if (failedChecksWithDetails.length > 0) {
          console.log(`
${theme.blue("Detailed Error Information")}
`);
          for (const check of failedChecksWithDetails) {
            console.log(theme.yellow(`${check.name}:`));
            console.log(theme.gray(check.details.split(`
`).map((line) => `  ${line}`).join(`
`)));
            console.log("");
          }
        }
      }
      console.log(`
${allPassed ? theme.green("All checks passed!") : theme.yellow("Some checks failed")}
`);
      return {
        success: true,
        data: {
          checks: context.checks,
          allPassed
        }
      };
    } catch (error) {
      return {
        success: false,
        error: {
          code: "UNKNOWN_ERROR",
          message: error instanceof Error ? error.message : String(error),
          recoverable: false
        }
      };
    }
  }
}

export { DoctorHandler };

//# debugId=FB405D19691F8C2E64756E2164756E21
