// @_davideast/stitch-mcp/dist/chunk-wa64nz8b.js
import"./chunk-9wyra8hs.js";

// src/commands/serve/handler.ts
class ServeHandler {
  client;
  constructor(client) {
    this.client = client;
  }
  async execute(projectId) {
    try {
      const project = await this.client.callTool("get_project", {
        name: `projects/${projectId}`
      });
      const screensResult = await this.client.callTool("list_screens", {
        projectId
      });
      const screens = (screensResult.screens || []).filter((screen) => !!screen.htmlCode?.downloadUrl).map((screen) => {
        const screenId = screen.name?.split("/screens/")[1] || screen.name;
        return {
          screenId,
          title: screen.title || screenId,
          codeUrl: screen.htmlCode.downloadUrl
        };
      });
      screens.sort((a, b) => a.title.localeCompare(b.title));
      return {
        success: true,
        projectId,
        projectTitle: project.title || projectId,
        screens
      };
    } catch (error) {
      return {
        success: false,
        error: error.message || String(error)
      };
    }
  }
  getClient() {
    return this.client;
  }
}
export {
  ServeHandler
};

//# debugId=E467E501360A240864756E2164756E21
