// @_davideast/stitch-mcp/dist/chunk-qv44tmn6.js
import {
  require_lib
} from "./chunk-w8q7nsm7.js";
import {
  __toESM
} from "./chunk-9wyra8hs.js";

// src/commands/site/utils/ProjectSyncer.ts
class ProjectSyncer {
  client;
  constructor(client) {
    this.client = client;
  }
  async fetchManifest(projectId) {
    const response = await this.client.callTool("list_screens", {
      projectId
    });
    return response.screens || [];
  }
  async fetchContent(url) {
    const maxRetries = 4;
    for (let attempt = 0;attempt <= maxRetries; attempt++) {
      const response = await fetch(url);
      if (response.ok)
        return await response.text();
      if (response.status === 429 && attempt < maxRetries) {
        const backoff = Math.min(1000 * 2 ** attempt, 8000);
        await new Promise((r) => setTimeout(r, backoff));
        continue;
      }
      throw new Error(`Failed to fetch content: ${response.statusText}`);
    }
    throw new Error(`Failed to fetch content after ${maxRetries + 1} attempts`);
  }
}

// src/lib/services/site/SiteService.ts
var import_fs_extra = __toESM(require_lib(), 1);
import path from "path";

// node_modules/yocto-queue/index.js
class Node {
  value;
  next;
  constructor(value) {
    this.value = value;
  }
}

class Queue {
  #head;
  #tail;
  #size;
  constructor() {
    this.clear();
  }
  enqueue(value) {
    const node = new Node(value);
    if (this.#head) {
      this.#tail.next = node;
      this.#tail = node;
    } else {
      this.#head = node;
      this.#tail = node;
    }
    this.#size++;
  }
  dequeue() {
    const current = this.#head;
    if (!current) {
      return;
    }
    this.#head = this.#head.next;
    this.#size--;
    if (!this.#head) {
      this.#tail = undefined;
    }
    return current.value;
  }
  peek() {
    if (!this.#head) {
      return;
    }
    return this.#head.value;
  }
  clear() {
    this.#head = undefined;
    this.#tail = undefined;
    this.#size = 0;
  }
  get size() {
    return this.#size;
  }
  *[Symbol.iterator]() {
    let current = this.#head;
    while (current) {
      yield current.value;
      current = current.next;
    }
  }
  *drain() {
    while (this.#head) {
      yield this.dequeue();
    }
  }
}

// node_modules/p-limit/index.js
function pLimit(concurrency) {
  let rejectOnClear = false;
  if (typeof concurrency === "object") {
    ({ concurrency, rejectOnClear = false } = concurrency);
  }
  validateConcurrency(concurrency);
  if (typeof rejectOnClear !== "boolean") {
    throw new TypeError("Expected `rejectOnClear` to be a boolean");
  }
  const queue = new Queue;
  let activeCount = 0;
  const resumeNext = () => {
    if (activeCount < concurrency && queue.size > 0) {
      activeCount++;
      queue.dequeue().run();
    }
  };
  const next = () => {
    activeCount--;
    resumeNext();
  };
  const run = async (function_, resolve, arguments_) => {
    const result = (async () => function_(...arguments_))();
    resolve(result);
    try {
      await result;
    } catch {}
    next();
  };
  const enqueue = (function_, resolve, reject, arguments_) => {
    const queueItem = { reject };
    new Promise((internalResolve) => {
      queueItem.run = internalResolve;
      queue.enqueue(queueItem);
    }).then(run.bind(undefined, function_, resolve, arguments_));
    if (activeCount < concurrency) {
      resumeNext();
    }
  };
  const generator = (function_, ...arguments_) => new Promise((resolve, reject) => {
    enqueue(function_, resolve, reject, arguments_);
  });
  Object.defineProperties(generator, {
    activeCount: {
      get: () => activeCount
    },
    pendingCount: {
      get: () => queue.size
    },
    clearQueue: {
      value() {
        if (!rejectOnClear) {
          queue.clear();
          return;
        }
        const abortError = AbortSignal.abort().reason;
        while (queue.size > 0) {
          queue.dequeue().reject(abortError);
        }
      }
    },
    concurrency: {
      get: () => concurrency,
      set(newConcurrency) {
        validateConcurrency(newConcurrency);
        concurrency = newConcurrency;
        queueMicrotask(() => {
          while (activeCount < concurrency && queue.size > 0) {
            resumeNext();
          }
        });
      }
    },
    map: {
      async value(iterable, function_) {
        const promises = Array.from(iterable, (value, index) => this(function_, value, index));
        return Promise.all(promises);
      }
    }
  });
  return generator;
}
function validateConcurrency(concurrency) {
  if (!((Number.isInteger(concurrency) || concurrency === Number.POSITIVE_INFINITY) && concurrency > 0)) {
    throw new TypeError("Expected `concurrency` to be a number from 1 and up");
  }
}

// src/lib/services/site/SiteService.ts
class SiteService {
  static toUIScreens(screens) {
    return screens.filter((s) => s.htmlCode && s.htmlCode.downloadUrl).map((s) => ({
      id: s.name,
      title: s.title,
      downloadUrl: s.htmlCode.downloadUrl,
      status: "ignored",
      route: ""
    }));
  }
  static async generateSite(config, htmlContent, assetGateway, outputDir = ".") {
    await import_fs_extra.default.ensureDir(path.join(outputDir, "src/pages"));
    await import_fs_extra.default.ensureDir(path.join(outputDir, "src/layouts"));
    await import_fs_extra.default.ensureDir(path.join(outputDir, "public/assets"));
    const pkgJson = {
      name: "stitch-site",
      type: "module",
      version: "0.0.1",
      scripts: {
        dev: "astro dev",
        start: "astro dev",
        build: "astro build",
        preview: "astro preview",
        astro: "astro"
      },
      dependencies: {
        astro: "^5.0.0"
      }
    };
    await import_fs_extra.default.writeJson(path.join(outputDir, "package.json"), pkgJson, { spaces: 2 });
    const astroConfig = `import { defineConfig } from 'astro/config';
export default defineConfig({});`;
    await import_fs_extra.default.writeFile(path.join(outputDir, "astro.config.mjs"), astroConfig);
    const layout = `---
interface Props {
	title: string;
}

const { title } = Astro.props;
---

<!doctype html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<meta name="description" content="Astro description" />
		<meta name="viewport" content="width=device-width" />
		<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
		<meta name="generator" content={Astro.generator} />
		<title>{title}</title>
	</head>
	<body>
		<slot />
	</body>
</html>
`;
    await import_fs_extra.default.writeFile(path.join(outputDir, "src/layouts/Layout.astro"), layout);
    const limit = pLimit(10);
    const tasks = config.routes.map((route) => limit(async () => {
      if (route.status !== "included")
        return;
      const html = htmlContent.get(route.screenId);
      if (!html) {
        console.warn(`No HTML content found for screen ${route.screenId}`);
        return;
      }
      const { html: rewrittenHtml, assets } = await assetGateway.rewriteHtmlForBuild(html);
      const assetsDir = path.join(outputDir, "public/assets");
      for (const asset of assets) {
        await assetGateway.copyAssetTo(asset.url, path.join(assetsDir, asset.filename));
      }
      let filePath = route.route;
      if (filePath === "/") {
        filePath = "index";
      } else {
        if (filePath.startsWith("/"))
          filePath = filePath.substring(1);
      }
      const fullPath = path.join(outputDir, "src/pages", `${filePath}.astro`);
      await import_fs_extra.default.ensureDir(path.dirname(fullPath));
      await import_fs_extra.default.writeFile(fullPath, rewrittenHtml);
    }));
    await Promise.all(tasks);
  }
  static slugify(text) {
    return text.toString().toLowerCase().replace(/\s+/g, "-").replace(/[^\w\-]+/g, "").replace(/\-\-+/g, "-").replace(/^-+/, "").replace(/-+$/, "");
  }
}

export { ProjectSyncer, pLimit, SiteService };

//# debugId=48437CEB05AF8D1664756E2164756E21
